src/test/java/core/basesyntax/service/impl/DataConverterImplTest.java

```diff
@@ -0,0 +1,101 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class DataConverterImplTest {
+    private static final String APPLE = "Apple";
+    private static final String BANANA = "Banana";
+    private static final String ORANGE = "Orange";
+    private static final String GRAPES = "Grapes";
+    private DataConverter dataConverter;
+
+    @BeforeEach
+    void setUp() {
+        dataConverter = new DataConverterImpl();
+    }
+
+    @Test
+    public void convertToTransaction_emptyData() {
+        List<String> emptyData = Collections.emptyList();
+        List<FruitTransaction> transactions = dataConverter.convertToTransaction(emptyData);
+        assertTrue(transactions.isEmpty());
+    }
+
+    @Test
+    public void convertToTransaction_nullData() {
+        assertThrows(NullPointerException.class,
+                () -> dataConverter.convertToTransaction(null));
+    }
+
+    @Test
+    public void convertToTransaction_invalidOperation() {
+        DataConverterImpl dataConverter = new DataConverterImpl();
+        List<String> data = Arrays.asList(
+                "INVALID,Apple,10",
+                "PURCHASE,Banana,abc",
+                "RETURN,Orange"
+        );
+        assertThrows(IllegalArgumentException.class,
+                () -> dataConverter.convertToTransaction(data));
+    }
+
+    @Test
+    void convertReport_inputMoreThanThreeParameters_notOk() {
+        List<String> wrong = List.of("1,2,3,4", "1,2,3,4,5");
+        Assertions.assertThrows(RuntimeException.class, () -> dataConverter
+                .convertToTransaction(wrong));
+    }
+
+    @Test
+    public void convertToTransaction_ValidData() {
```

Comment:
method naming issues. Please follow the same convention. Also please try to make this method shorter if possible

Reply:
fix the same issue in all places